lib/LedgerSMB/Admin/Command/user.pm

```
@@ -0,0 +1,582 @@
+    $emp->{middle_name} = $middle_name if $middle_name;
+    $emp->{role} = $role if $role;
+    $emp->{sales} = $is_sales if $is_sales;
+    $emp->{salutation} = $salutation if $salutation;
+    $emp->{ssn} = $ssn if $ssn;
+    $emp->{start_date} = $start_date if $start_date;
+
+    $emp->save
+        if $country || $dob || $employeenumber || $end_date || $first_name
+        || $is_manager || $last_name || $manager || $middle_name || $role
+        || $is_sales || $salutation || $ssn || $start_date;
+
+    # Add permissions
+    return 1 if $permission && !$self->_add_permissions($dbh, $user);
+    # Remove permissions
+    return 1 if $no_permission && !$self->_remove_permissions($dbh, $user);
+
+    $dbh->commit if ! $dbh->{AutoCommit};
+    $dbh->disconnect;
+    $self->logger->warn('User successfully changed');
+
+    return 0;
+}
+
+sub create {
+    my ($self, $dbh, $options, @args) = @_;
+
+    my %options = ();
+    Getopt::Long::Configure(qw(bundling require_order));
+    GetOptionsFromArray(\@args, \%options, $self->_option_spec('create'));
+    if (!$username || !$password){
+        $self->logger->error('Missing username or password');
+        return 1;
+    }
+
+    if ($self->_get_user($dbh)) {
+        $self->logger->error("User '$username' already exists");
+        $dbh->rollback;
+        return 1;
+    }
+    my $emp = $self->_create_employee(dbh => $dbh);
+    if (!$emp) {
+        $self->logger->error('Invalid employee');
+        $dbh->rollback;
+        return 1;
+    }
+    my $user = $self->_create_user(
+        dbh => $dbh,
+        entity_id => $emp->entity_id,
+        username => $username,
+        password => $password
+    );
+    if (!$user) {
+        $self->logger->error('Invalid user');
+        $dbh->rollback;
+        return 1;
+    }
+    #TODO: Fix validity
+    $dbh->do(qq(ALTER USER "$username" VALID UNTIL 'infinity'));
+
+    # Add permissions
+    return 1 if $permission && !$self->_add_permissions($dbh, $user);
```

This prevents adding and removing permissions in a single invocation. Is that intentional?

---

No, this fails early when permissions were given but aren't valid.
Same for no-permissions